toolkit/tools/imagegen/configuration/packagerepo.go

```diff
@@ -0,0 +1,203 @@
+}
+
+func writeAdditionalFieldstoRepoFile(fileName string) (err error) {
+	const (
+		gpgKey    = "gpgkey=file:///etc/pki/rpm-gpg/MICROSOFT-RPM-GPG-KEY file:///etc/pki/rpm-gpg/MICROSOFT-METADATA-GPG-KEY\n"
+		enable    = "enabled=1\n"
+		gpgCheck  = "gpgcheck=0\n"
+		skip      = "skip_if_unavailable=True\n"
+		sslVerify = "sslverify=0\n"
+	)
+
+	additionalFields := gpgKey + enable + gpgCheck + skip + sslVerify
+	err = file.Append(additionalFields, fileName)
+	if err != nil {
+		logger.Log.Errorf("Failed to write additional fields")
+	}
+
+	return
+}
+
+func createCustomRepoFile(fileName string, packageRepo PackageRepo) (err error) {
+	err = file.Create(fileName, 0644)
+	if err != nil {
+		logger.Log.Errorf("Error creating file %s", fileName)
+		return
+	}
+
+	// Write the repo identifier field
+	repoId := "[" + packageRepo.Name + "]\n"
+	err = file.Append(repoId, fileName)
+	if err != nil {
+		logger.Log.Errorf("Failed to write repo identifier %s", repoId)
+		return
+	}
+
+	// Write the repo Name field
+	repoName := "name=" + packageRepo.Name + " $releasever $basearch\n"
+	err = file.Append(repoName, fileName)
+	if err != nil {
+		logger.Log.Errorf("Failed to write repo name %s", repoName)
+		return
+	}
+
+	// Write the repo URL field
+	repoUrl := "baseurl=" + packageRepo.BaseUrl + "\n"
+	err = file.Append(repoUrl, fileName)
+	if err != nil {
+		logger.Log.Errorf("Failed to write repo URL %s", repoUrl)
+		return
+	}
+
+	err = writeAdditionalFieldstoRepoFile(fileName)
+	return
+}
+
+func createCustomPackageRepo(installChroot *safechroot.Chroot, packageRepo PackageRepo, repoFileDir string) (err error) {
+
+	dstRepoPath := repoFileDir + packageRepo.Name + ".repo"
+
+	// Create repo file
+	err = createCustomRepoFile(dstRepoPath, packageRepo)
+	if err != nil {
+		return
+	}
+
+	// Check the repo file needs to be installed in the image
+	if packageRepo.Install {
+		installRepoFile := filepath.Join(installChroot.RootDir(), dstRepoPath)
+		err = file.Copy(dstRepoPath, installRepoFile)
+	}
+
+	return
+}
+
+// UpdatePackageRepo creates additional repo files specified by image configuration
+// and returns error if the operation fails
+func UpdatePackageRepo(installChroot *safechroot.Chroot, config SystemConfig) (err error) {
+	const (
+		repoFileDir   = "/etc/yum.repos.d/"
+		localRepoFile = "/etc/yum.repos.d/mariner-iso.repo"
+		squashErrors  = false
+	)
+
+	if exists, ferr := file.DirExists(repoFileDir); ferr != nil {
+		logger.Log.Errorf("Error accessing repo file directory.")
+		err = ferr
+		return
+	} else if !exists {
+		err = fmt.Errorf("%s: no such directory", repoFileDir)
+		return
+	}
+
+	// Remove mariner-iso.repo
+	if exists, ferr := file.PathExists(localRepoFile); ferr != nil {
+		logger.Log.Errorf("Error accessing /etc/yum.repos.d/mariner-iso.repo")
+		err = ferr
+		return
+	} else if !exists {
+		err = fmt.Errorf("%s: no such file or directory", localRepoFile)
+		return
+	}
+
+	err = shell.ExecuteLive(squashErrors, "rm", localRepoFile)
```

Comment:
What happens if the user wants to use a combination of packages pre-installed to the ISO and a few from their custom repository?

Reply:
We have discussed with our customer about this and they said that they don't want to pull packages from local repo during the installation process.

Reply:
I'm also of the opinion that we would want the flexibility to do both, but that feature can wait for a future update.  (Just file a work item on it so we don't forget)

Reply:
Sounds good!